hpx/util/detail/pack_traversal_async_impl.hpp

```diff
@@ -514,8 +553,7 @@ namespace util {
         template <typename Frame, typename State>
         void resume_traversal_callable<Frame, State>::operator()()
         {
-            auto hierarchy =
-                util::tuple_cat(util::make_tuple(frame_), std::move(state_));
+            auto hierarchy = util::tuple_cat(util::make_tuple(frame_), state_);
```

Comment:
Every `resume_traversal_callable` instance should only be called once, so the move should be valid here.
Maybe we could guard this through an assertion.

Which issue did occur with this move?

Reply:
Frankly, I was not sure whether this may have caused my issues. I will revert this change and add an assertion making sure this doesn't cause any problems.

Reply:
BTW, the same might be true for this change: https://github.com/STEllAR-GROUP/hpx/pull/3001/files#diff-a33368bac14c4ebae471f68e8cb9da76R381